### ISSUE TITLE
`run` prints value with bogus type

### ISSUE THREAD

**Issue by byorgey** [swarm-game/swarm#328]
**Describe the bug**
Calling `run` on certain files causes the result of executing the file to be printed along with a unit type, regardless of the actual type of the result.

**To Reproduce**
In `true.sw`:
```
true
```
Now start swarm and at the REPL, type `run("true.sw")`  This will print out
```
true : ()
```

**Expected behavior**
It should result in
```
true : bool
```
(or perhaps it shouldn't print anything at all?).

**Comment by byorgey** [comment]
Another odd thing which seems related: 
```
> if true {move; move} {move}
> if true {x <- move; move} {move}
() : ()
```
Both execute normally, but in the first case nothing is printed, whereas in the second case `() : ()` is printed.  I think it has something to do with binds generating a `VResult`...

**Comment by byorgey** [comment]
When an expression is entered at the REPL, we first store its type by setting the `replStatus` to `REPLWorking ty Nothing`.  Then once the CESK machine finishes we take the value that results and print it out along with the type.  So the type of `run("foo.sw")` is `cmd ()` (there is a special case that strips off the `cmd` before printing the type).  But the problem is that `run("foo.sw")` basically replaces itself so that we actually evaluate the contents of the file `foo.sw`; but the type in the `replStatus` never gets updated.  

...I guess this means `run` actually breaks type preservation!  The more I think about it the less I like `run` and the way it currently works.  It's expedient but not very well-motivated theoretically.

**Comment by byorgey** [comment]
I also encountered `() : ()` being printed when defining a function that called `build` and did a few other things, then calling that function at the REPL prompt.

**Comment by byorgey** [comment]
To be clear at this point, I am keeping this open so we remember that it's a problem, but I expect it will be completely and properly solved by #495, so this issue is not worth working on in and of itself.

**Comment by xsebek** [comment]
Fun idea, would it make more sense for run to have type `cmd a`?

Then you could at least use the return value at your own risk.

**Comment by byorgey** [comment]
I don't see how that would be better.  It would still be wrong, and you can already use the return value at your own risk.

**Comment by xsebek** [comment]
@byorgey no, you can only use it as a result of the whole computation (e.g. in objective #828), not in the same term.

Consider this file `r.sw`:
```Java
b <- random 1;
if (b == 0) {
    r <- run "r.sw";
    return $ r + 1; // <-- this r does not type check, because it is unit
} {
    return 0;
}
```

**Comment by byorgey** [comment]
Ah, good point.  Still, I don't really see the benefit.  I'd rather just do it properly and implement #495.  In fact that is one of the next things I was thinking of working on.

**Comment by byorgey** [comment]
Also related is #1087 which will hopefully get rid of all the `VResult` machinery.

**Comment by byorgey** [comment]
Note, #1928 seems to have gotten rid of the issue with `() : ()` being printed.  I think that was unrelated.  However, running `true.sw` still results in `it0 : Unit = true`.
